### PR TITLE
[glib] Support cross-compiling Objective C code using Meson

### LINKED ISSUE #20341: glib:arm64-osx build failure
**Host Environment**
-   OS: [macOS 11.6]
-   Compiler: Apple clang 12.0.5

**To Reproduce**
`./vcpkg install --triplet=arm64-osx openimageio`

**Failure logs** 
```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:127 (message):
    Command failed: /Library/Frameworks/Python.framework/Versions/3.7/bin/python3 /Users/mark/Projects/github/vcpkg/downloads/tools/meson/meson-aeda7f249c4a5dbbecc52e44f382246a2377b5b0/meson.py -Dbuild_tests=false -Dinstalled_tests=false -Dselinux=disabled -Dinternal_pcre=false --buildtype plain --backend ninja --wrap-mode nodownload --cross /Users/mark/Projects/github/vcpkg/buildtrees/glib/meson-cross-arm64-osx.log --native /Users/mark/Projects/github/vcpkg/scripts/buildsystems/meson/none.txt --default-library static --libdir lib --cross /Users/mark/Projects/github/vcpkg/buildtrees/glib/meson-cross-arm64-osx-debug.log -Ddebug=true --prefix /Users/mark/Projects/github/vcpkg/packages/glib_arm64-osx/debug --includedir ../include -Dcmake_prefix_path=['/Users/mark/Projects/github/vcpkg/installed/arm64-osx/debug','/Users/mark/Projects/github/vcpkg/installed/arm64-osx'] /Users/mark/Projects/github/vcpkg/buildtrees/glib/src/glib-2-b84205c786.clean
    Working Directory: /Users/mark/Projects/github/vcpkg/buildtrees/glib/arm64-osx-dbg
    Error code: 1
    See logs for more information:
      /Users/mark/Projects/github/vcpkg/buildtrees/glib/config-arm64-osx-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_meson.cmake:512 (vcpkg_execute_required_process)
  ports/glib/portfile.cmake:42 (vcpkg_configure_meson)
  scripts/ports.cmake:140 (include)


Error: Building package glib:arm64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: glib:arm64-osx
  Vcpkg version: 2021-09-10-unknownhash

Additionally, attach any relevant sections from the log files above.
```
-Please attach any additional failure logs mentioned in the console output.

[config-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/7229663/config-x64-osx-dbg-out.log)

The relevant part is

```
Compiler for language objc for the build machine not found.

../src/glib-2-b84205c786.clean/meson.build:733:2: ERROR: 'objc' compiler binary not defined in cross or native file

A full log can be found at /Users/mark/Projects/github/vcpkg/buildtrees/glib/arm64-osx-dbg/meson-logs/meson-log.txt
```

Here is the log file mentioned in that message.

[meson-log.txt](https://github.com/microsoft/vcpkg/files/7229667/meson-log.txt)

The relevant part of it:

```
Compiler for language objc for the build machine not found.

../src/glib-2-b84205c786.clean/meson.build:733:2: ERROR: 'objc' compiler binary not defined in cross or native file
```

doesn't add anything useful.

**Additional context**
- Vcpkg version: 2021-09-10-unknownhash

### LINKED ISSUE #26147: Meson build is missing ObjectiveC support
**Describe the bug**
Cross Compile to x64 mac from arm64-mac fails due to lack of support in meson native config

**Environment**
- OS: macOS
- Compiler: Xcode 13.x

**To Reproduce**
``vcpkg install glib --triplet=x64-osx`` on a AArch64 Mac
2. See error

**Failure logs**
[config-x64-osx-dbg-meson-log.txt.log](https://github.com/microsoft/vcpkg/files/9249054/config-x64-osx-dbg-meson-log.txt.log)
[config-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/9249055/config-x64-osx-dbg-out.log)

**Additional context**
Core problem is this error during meson configuration:

```
Header <dlfcn.h> has symbol "RTLD_NEXT" : YES 
Message: Checking whether to use statfs or statvfs .. statfs
Compiler for language objc for the build machine not found.

../src/2.72.3-eda9959d65.clean/meson.build:799:2: ERROR: 'objc' compiler binary not defined in cross or native file
```

Which I suspect is due to ``z_vcpkg_meson_generate_cross_file`` in ``vcpkg_configure_meson.cmake`` but I don't know any meson to submit my own patch

I'm currently using vcpkg @ 5bb332b432e0fa3899de1a2ef8fc4c10c435408d environment listed below:

```
➜  vcpkg git:(master) ✗ sw_vers
ProductName:	macOS
ProductVersion:	12.5
BuildVersion:	21G72
➜  vcpkg git:(master) ✗ arch
arm64
➜  vcpkg git:(master) ✗ meson --version
0.63.0
➜  vcpkg git:(master) ✗ cmake --version
cmake version 3.23.1

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

### PR DESCRIPTION
**Pull request by qmfrederik**
The glib port uses Objective C code and the Meson build system. You'll end up cross compiling glib when compiling for x64 osx on an arm64 osx machine, or vice versa.

For this to work, an Objective C compiler needs to be set in the Meson cross file.

vcpkg generates this file based on the
`scripts/buildsystems/meson/meson.template.in` template, and expects the `MESON_OBJC` CMake variable to be set.

That's the job of `z_vcpkg_meson_set_proglist_variables` in `scripts/cmake/vcpkg_configure_meson.cmake`.  It will only generate data for an Objective C compiler if:

- The `Languages` argument includes `OBJC`
- CMake has detected a Objective C compiler, and has set `VCPKG_DETECTED_CMAKE_OBJC_COMPILER`

To make this work, this patch:
- Enables the OBJC and OBJCXX language on Apple platforms
- Updates the call to `vcpkg_configure_meson` in the glib port file to include the OBJC OBJCXX languages.

Fixes https://github.com/microsoft/vcpkg/issues/26147 Fixes https://github.com/microsoft/vcpkg/issues/20341 Based on https://github.com/microsoft/vcpkg/pull/26259 Based on https://github.com/TechSmith/vcpkg/commit/27dee354e4a4b267b8ad963870981b16f6ada6c5

Tested on an arm64 mac to compile glib for x64 macos

### CODE REVIEW

File: scripts/detect_compiler/CMakeLists.txt [hunk 1]
```diff
@@ -10,10 +10,21 @@ endif()
 
 enable_language(C)
 enable_language(CXX)
+if(APPLE)
+    enable_language(OBJC)
+    enable_language(OBJCXX)
+endif()
```
Comment (Neumann-A) [inline scripts/detect_compiler/CMakeLists.txt:16]: @ras0219-msft: I think this needs to be configurable somehow.

File: scripts/get_cmake_vars/CMakeLists.txt [hunk 2]
```diff
@@ -11,6 +11,10 @@ else()
 endif()
 
 set(VCPKG_LANGUAGES "C;CXX" CACHE STRING "Languages to enables for this project")
+if(APPLE)
+    list(APPEND VCPKG_LANGUAGES "OBJC")
+    list(APPEND VCPKG_LANGUAGES "OBJCXX")
+endif()
```
Comment (Neumann-A) [inline scripts/get_cmake_vars/CMakeLists.txt:17]: This always enable OBJC and OBJCXX on macosx running unnecessary compiler checks. You can use `list(APPEND VCPKG_CMAKE_CONFIGURE_OPTIONS "-DVCPKG_LANGUAGES=C;CXX;OBJC;OBJCXX")` from the portfile or in `vcpkg_configure_meson` to enable additional languages.

File: scripts/detect_compiler/CMakeLists.txt (hunk 1 above)
Comment (qmfrederik) [reply scripts/detect_compiler/CMakeLists.txt:16]: @Neumann-A If we don't want enable ObjC on macOS by default, would a better alternative be to add support for ObjC to the port files that need it?  E.g. in this case I just add it to `ports/glib/portfile.cmake`?

Comment (Neumann-A) [reply scripts/detect_compiler/CMakeLists.txt:16]: Probably yes. Other ports do that on a port level basis with `ASM` and `Fortran`

**Comment by qmfrederik** [comment]
@Neumann-A Thank you for your feedback! I was able to simplify the PR a bit, and confirmed it works on my M2 Mac when cross-compiling for x64 Mac (something which fails without this patch)

**Pull request merged**